documentation/source/ObjectDetection.md

```diff
@@ -0,0 +1,312 @@
+
+```python
+from super_gradients.training.models.detection_models.yolo_base import YoloPostPredictionCallback
+
+
+post_prediction_callback = YoloPostPredictionCallback(conf=0.001, iou=0.6)
+```
+
+### Visualization
+
+Visualization of the model predictions is a very important part of the training process for any computer vision task. 
+By visualizing the predicted boxes, developers and researchers can identify errors or inaccuracies in the model's output and adjust the model's architecture or training data accordingly.
+
+SuperGradients provide an implementation of [DetectionVisualizationCallback](https://docs.deci.ai/super-gradients/docstring/training/utils/#training.utils.callbacks.callbacks.DetectionVisualizationCallback). 
+You can use this callback in your training pipeline to visualize predictions during training. For this, just add it to `training_hyperparams.phase_callbacks` in your yaml.
+During training, the callback will generate a visualization of the model predictions and save it to the TensorBoard or Weights & Biases depending on which logger you
+are using (Default is Tensorboard). 
+
+If you would like to do the visualization outside of training you can use `DetectionVisualization` class as follows:
+```python
+import torch
+import numpy as np
+
+from super_gradients.training import models
+from super_gradients.training.utils.detection_utils import DetectionVisualization
+from super_gradients.training.datasets.datasets_conf import COCO_DETECTION_CLASSES_LIST
+
+
+def my_undo_image_preprocessing(im_tensor: torch.Tensor) -> np.ndarray:
+    im_np = im_tensor.cpu().numpy()
+    im_np = im_np[:, ::-1, :, :].transpose(0, 2, 3, 1)
+    im_np *= 255.0
+    return np.ascontiguousarray(im_np, dtype=np.uint8)
+
+
+model = models.get("yolox_s", pretrained_weights="coco", num_classes=80)
+imgs, targets = iter(train_dataloader).__next__()
+preds = YoloPostPredictionCallback(conf=0.1, iou=0.6)(model(imgs))
+DetectionVisualization.visualize_batch(imgs, preds, targets, batch_name='train', class_names=COCO_DETECTION_CLASSES_LIST,
+                                       checkpoint_dir='/path/for/saved_images/', gt_alpha=0.5,
+                                       undo_preprocessing_func=my_undo_image_preprocessing)
+```
+The function you pass as `undo_preprocessing_func` will define how to undo dataset transforms and return the image back into its initial formal (BGR, uint8).
+This also allows you to test the correctness of your dataset implementation, since it saves images after they go through transforms. This may be especially useful for a train set with heavy augmentation transforms. You can see both the predictions and the ground truth, and give the ground truth box the desired opacity. 
+
+The saved train image for a dataset with a mosaic transform should look something like this:
+
+![train_24](images/train_24.jpg)
+
+## How to connect your own dataset
```

Comment:
before this. please add a code example of how everything you wrote above falls into place.
training params with the metric, the callback, the data-loaders

Reply:
Added more story in 2653fbb